aten/src/ATen/native/Gesv.cpp

```diff
@@ -43,6 +43,12 @@ template<> void lapackGesv<double>(
 }
 #endif
 
+static inline bool isTransposeContiguous(Tensor& self) {
+ return self.dim() == 2 &&
```

Comment:
you can optimize this further by allowing dimension with size `<= 1` to have arbitrary stride

Reply:
Here is what I meant: If a dimension has size `<= 1`, it can have arbitrary stride at that dimension and still be contiguous because the index at that dimension is at most `0`.

Reply:
It's up to you whether to add more boolean statements :) This is fine as is by me.